docs/pages/9 - Contrib Modules.md

```diff
@@ -165,76 +165,68 @@ object app extends ScalaModule with TwirlModule {
+`mill.playlib.routesModule` trait when defining your module. 
+
+ 
+```scala
+import $ivy.`com.lihaoyi::mill-contrib-playlib:0.3.3`,  mill.playlib._
+object app extends RouterModule {
+// ...
+} 
+``` 
+
 #### Configuration options
 
-* `def twirlVersion: T[String]` (mandatory) - the version of the twirl compiler to use, like "1.3.15"
-* `def twirlAdditionalImports: Seq[String] = Nil` - the additional imports that will be added by twirl compiler to the top of all templates
-* `def twirlConstructorAnnotations: Seq[String] = Nil` - annotations added to the generated classes' constructors (note it only applies to templates with `@this(...)` constructors) 
-* `def twirlCodec = Codec(Properties.sourceEncoding)` - the codec used to generate the files (the default is the same sbt plugin uses) 
-* `def twirlInclusiveDot: Boolean = false`  
+  * `def playVersion: T[String]` (mandatory) - The version of play to use to compile the routes file.
+  * `def scalaVersion: T[String]` - The scalaVersion in use in your project. (Defaults to the default scalaVersion for a given play version.)
+  * `def routesFile: T[PathRef]` - The path to the main routes file. (Defaults to the standard play path : `conf/routes`.)
+  * `def routesAdditionalImport: Seq[String]` - Additional imports to use in the generated routers. (Defaults to `Seq("controllers.Assets.Asset", "play.libs.F")`
+  * `def generateForwardsRouter: Boolean = true` - Enables the forward router generation.
+  * `def generateReverseRouter: Boolean = true` - Enables the reverse router generation.
+  * `def namespaceReverseRouter: Boolean = false` - Enables the namespacing of reverse routers.
+  * `def generatorType: RouteCompilerType = RouteCompilerType.InjectedGenerator` - The routes compiler type, one of RouteCompilerType.InjectedGenerator or RouteCompilerType.StaticGenerator
   
 #### Details
 
-The following filesystem layout is expected:
+The following filesystem layout is expected by default:
 
 ```text
 build.sc
-app/
-    views/
-        view1.scala.html
-        view2.scala.html
+conf/
+    routes
 ```
 
-`TwirlModule` adds the `compileTwirl` task to the module:
+`RouterModule` adds the `compileRouter` task to the module:
 ```
-mill app.compileTwirl
+mill app.compileRouter
 ```
 
 (it will be automatically run whenever you compile your module)
 
-This task will compile `*.scala.html` templates (and others, like `*.scala.txt`) into the `out/app/compileTwirl/dest` 
+This task will compile `routes` templates into the `out/app/compileRouter/dest` 
 directory. This directory must be added to the generated sources of the module to be compiled and made accessible from the rest of the code:
 ```scala
-object app extends ScalaModule with TwirlModule {
-  def twirlVersion = "1.3.15"
-  def generatedSources = T{ Seq(compileTwirl().classes) }
+object app extends ScalaModule with RouterModule {
+  def playVersion= "2.7.0"
+  def generatedSources = T{ Seq(compileRouter().classes) }
```

Comment:
Can't we provide that behavior by default? If not, we should at least suggest a cumulative addition.
 ```suggestion
  def generatedSources = T{ super.generatedSources() ++ Seq(compileRouter().classes) }
```

Reply:
We can probably provide that by default, I followed the structure of the TwirlModule and didn't add the behaviour by default. I don't mind adding it.

Reply:
With "behavior" I meant the addition of the `generatedSources`. It looks like as they need to be added in all cases.

Reply:
Ups, commented at same time... What I wanted to say, if we always need to add those generatedSources, than we should provide a reasonable default.

Reply:
>With "behavior" I meant the addition of the `generatedSources`
I got that :) As I said I matched the way the TwirlModule did things, as far as I know this behavior would be needed for Twirl too.

Reply:
I'll add the default to the RouterModule. I guess I could open a separate PR to add it to the twirl module too :)